packages/google_sign_in/google_sign_in_android/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -335,20 +340,40 @@ public void init(
             throw new IllegalStateException("Unknown signInOption");
         }
 
-        // Only requests a clientId if google-services.json was present and parsed
-        // by the google-services Gradle script.
-        // TODO(jackson): Perhaps we should provide a mechanism to override this
-        // behavior.
-        int clientIdIdentifier =
-            context
-                .getResources()
-                .getIdentifier("default_web_client_id", "string", context.getPackageName());
+        // The clientId parameter is not supported on Android.
+        // Android apps are identified by their package name and the SHA-1 of their signing key.
+        // https://developers.google.com/android/guides/client-auth
+        // https://developers.google.com/identity/sign-in/android/start#configure-a-google-api-project
         if (!Strings.isNullOrEmpty(clientId)) {
-          optionsBuilder.requestIdToken(clientId);
-          optionsBuilder.requestServerAuthCode(clientId);
-        } else if (clientIdIdentifier != 0) {
-          optionsBuilder.requestIdToken(context.getString(clientIdIdentifier));
-          optionsBuilder.requestServerAuthCode(context.getString(clientIdIdentifier));
+          if (Strings.isNullOrEmpty(serverClientId)) {
+            Log.w(
+                "google_sing_in",
+                "clientId is not supported on Android and is interpreted as serverClientId. "
```

Comment:
nits: 
I don't think we should talk about the "history" in the warning message, we should make it as brief and clear as possible, maybe:
```
"clientId is not supported on Android and is interpreted as serverClientId."
"Use serverClientId instead to suppress this warning."
```

Reply:
Done